### PR TITLE
[IOS-607] Profile of a user - info screen

### PR DESCRIPTION
**Pull request by dwdmarkowski**
### Ticket
[IOS-607](https://netguru.atlassian.net/browse/IOS-607)

### Task Description
Implemented user's info screen

### CODE REVIEW

File: Inbbbox/Resources/es.lproj/Localizable.strings [hunk 1]
```diff
@@ -353,3 +356,11 @@
...
 "ShotCollectionCell.Comment" = "Comentando…";
 "ShotCollectionCell.Follow" = "Siguiendo al autor!";
+
+/********* ProfileInfoView *********/
+"ProfileInfoView.Followers" = "$Followers";
+"ProfileInfoView.Shots" = "$Shots";
+"ProfileInfoView.Following" = "Siguiendo";
+
+/********* TeamsCollectionHeaderView *********/
+"TeamsCollectionHeaderView.OnTeams" = "$On Teams";
```
Comment (Myrenkar) [inline Inbbbox/Resources/es.lproj/Localizable.strings]: Empty line needed.

File: Inbbbox/Resources/pl.lproj/Localizable.strings [hunk 2]
```diff
@@ -352,3 +355,11 @@
 "ShotCollectionCell.Bucket" = "Dodano do polubień i wiaderka!";
 "ShotCollectionCell.Comment" = "Komentowanie…";
 "ShotCollectionCell.Follow" = "Obserwujesz autora!";
+
+/********* ProfileInfoView *********/
+"ProfileInfoView.Followers" = "Followers"; //$
```
Comment (Myrenkar) [inline Inbbbox/Resources/pl.lproj/Localizable.strings]: what is purpose of this $?

File: Inbbbox/Resources/pt-PT.lproj/Localizable.strings [hunk 3]
```diff
@@ -352,3 +355,11 @@
...
 "ShotCollectionCell.Comment" = "Comentando…";
 "ShotCollectionCell.Follow" = "Está a seguir ao autor!";
+
+/********* ProfileInfoView *********/
+"ProfileInfoView.Followers" = "$Followers";
+"ProfileInfoView.Shots" = "$Shots";
+"ProfileInfoView.Following" = "Os autores seguidos";
+
+/********* TeamsCollectionHeaderView *********/
+"TeamsCollectionHeaderView.OnTeams" = "$On Teams";
```
Comment (Myrenkar) [inline Inbbbox/Resources/pt-PT.lproj/Localizable.strings]: New line needed.

File: Inbbbox/Source Files/Model/Team.swift [hunk 4]
```diff
@@ -41,6 +49,10 @@ extension Team: Mappable {
         case Username = "username"
         case Avatar = "avatar_url"
         case CreatedAt = "created_at"
+        case FollowersCount = "followers_count"
+        case FollowingsCount = "followings_count"
+        case Bio = "bio"
+        case Location = "location"
```
Comment (Myrenkar) [inline Inbbbox/Source Files/Model/Team.swift]: All this should be lowercased due to swift 3 convention.

File: Inbbbox/Source Files/Model/User.swift [hunk 5]
```diff
@@ -61,6 +76,10 @@ private extension User {
         case Avatar = "avatar_url"
         case ShotsCount = "shots_count"
         case AccountType = "type"
+        case FollowersCount = "followers_count"
+        case FollowingsCount = "followings_count"
+        case Bio = "bio"
+        case Location = "location"
```
Comment (Myrenkar) [inline Inbbbox/Source Files/Model/User.swift]: the same, all enum should be lowercased

File: Inbbbox/Source Files/Networking/Queries/TeamsQuery.swift [hunk 6]
```diff
@@ -0,0 +1,19 @@
+//
+//  TeamsQuery.swift
+//  Inbbbox
+//
+//  Copyright © 2016 Netguru Sp. z o.o. All rights reserved.
+//
+
+struct TeamsQuery: Query {
```
Comment (Myrenkar) [inline Inbbbox/Source Files/Networking/Queries/TeamsQuery.swift]: final maybe?

File: Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift [hunk 7]
```diff
@@ -12,13 +12,49 @@ import PromiseKit
...
+
+    /**
+     Provides next page of teams.
+
+     - Warning: You have to use any of provide... method first to be able to use this method.
+     Otherwise an exception will appear.
 
      - returns: Promise which resolves with users or nil.
      */
+    func teamsNextPage() -> Promise<[TeamType]?> {
```
Comment (Myrenkar) [inline Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift]: Strange way of documenting, you should use build-in  `cmd + alt + /`  combination

File: Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift [hunk 8]
```diff
@@ -12,13 +12,49 @@ import PromiseKit
...
+        }
+    }
+
+    /**
+     Provides team's members.
+
+     - parameter team: Team to get members for.
+
+    - returns: Promise which resolves with users or nil.
+    */
```
Comment (Myrenkar) [inline Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift]: The same here

File: Inbbbox/Source Files/Views/Controllers View/ProfileInfoView.swift [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    }()
+
+    private lazy var informationsStackView: UIStackView = { [unowned self] in
+        let stackView = UIStackView(
+            arrangedSubviews: [
+                self.locationView,
+                self.bioLabel,
+            ]
+        )
+        stackView.layoutMargins = UIEdgeInsetsMake(16, 0, 12, 0)
```
Comment (Myrenkar) [inline Inbbbox/Source Files/Views/Controllers View/ProfileInfoView.swift]: there are two 12 here, are they somehow connected?

File: Inbbbox/Source Files/Views/Custom/SeparatorView.swift [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+import UIKit
+
+final class SeparatorView: UIView {
+
+    private let axis: Axis
+    private let thickness: Double
+    private let color: UIColor
+
+    @available(*, unavailable, message: "Use init(frame:) instead")
```
Comment (Myrenkar) [inline Inbbbox/Source Files/Views/Custom/SeparatorView.swift]: It doesnt conform to init below. No frame is initialized

File: Inbbbox/Source Files/Views/UICollectionViewCell/TeamCollectionViewCell.swift [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    private func setupLayout() {
+        addSubview(logoImageView)
+        addSubview(nameLabel)
+
+        logoImageView.autoSetDimensions(to: CGSize(width: 45, height: 45))
+        logoImageView.autoAlignAxis(toSuperviewAxis: .horizontal)
+        logoImageView.autoPinEdge(toSuperviewEdge: .left, withInset: 24)
+        logoImageView.layer.cornerRadius = 22.5
```
Comment (Myrenkar) [inline Inbbbox/Source Files/Views/UICollectionViewCell/TeamCollectionViewCell.swift]: you should combine those 45 with 22.5 somehow to make it dependant.

File: Inbbbox/Resources/pl.lproj/Localizable.strings (hunk 2 above)
Comment (dwdmarkowski) [inline Inbbbox/Resources/pl.lproj/Localizable.strings]: We mark missing translations which will be provided in the future like this.

File: Inbbbox/Resources/es.lproj/Localizable.strings (hunk 1 above)
Comment (dwdmarkowski) [inline Inbbbox/Resources/es.lproj/Localizable.strings]: 👍

File: Inbbbox/Resources/pt-PT.lproj/Localizable.strings (hunk 3 above)
Comment (dwdmarkowski) [inline Inbbbox/Resources/pt-PT.lproj/Localizable.strings]: 👍

File: Inbbbox/Source Files/Model/Team.swift (hunk 4 above)
Comment (dwdmarkowski) [inline Inbbbox/Source Files/Model/Team.swift]: 👍

File: Inbbbox/Source Files/Model/User.swift (hunk 5 above)
Comment (dwdmarkowski) [inline Inbbbox/Source Files/Model/User.swift]: 👍

File: Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift (hunk 7 above)
Comment (dwdmarkowski) [inline Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift]: I wanted to fit project's convention ¯\_(ツ)_/¯ I'll keep it as it is ok?

File: Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift (hunk 8 above)
Comment (dwdmarkowski) [inline Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift]: I wanted to fit project's convention ¯_(ツ)_/¯ I'll keep it as it is ok?

File: Inbbbox/Source Files/Networking/Queries/TeamsQuery.swift (hunk 6 above)
Comment (dwdmarkowski) [inline Inbbbox/Source Files/Networking/Queries/TeamsQuery.swift]: "only classes and class members may be marked with 'final'"

File: Inbbbox/Source Files/Views/Controllers View/ProfileInfoView.swift (hunk 9 above)
Comment (dwdmarkowski) [inline Inbbbox/Source Files/Views/Controllers View/ProfileInfoView.swift]: Not really :(

File: Inbbbox/Source Files/Views/Custom/SeparatorView.swift (hunk 10 above)
Comment (dwdmarkowski) [inline Inbbbox/Source Files/Views/Custom/SeparatorView.swift]: 👍

File: Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift (hunk 7 above)
Comment (Myrenkar) [inline Inbbbox/Source Files/Providers/API Providers/APITeamsProvider.swift]: No problem 👍

**Pull request merged**